code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/LoadBalancerServiceInstanceStopPreListener.java

```diff
@@ -174,7 +176,7 @@ protected void preProcessEvent(EventVO<?> event, ProcessState state, ProcessInst
     }
 
     private List<Service> lookupAllActiveLBServices(Service targetService) {
-        List<Service> lbServices = new ArrayList<>();
+        Set<Service> lbServices = new HashSet<>();
```

Comment:
I would much prefer that we used service.Id to build the set than the service itself. We don't really guarantee that the thing that implements service has a well implemented hashCode and equals method.

Looking at the current implementation, it compares each field on the object. That is actually much more than we need or want.

You could use a map like `Map<Long, Service>` to rely on keys being unique and then use `map.values()` to get the services at the end.

Reply:
yep, my bad, should have used this first.